src/mappings/shared/token/setMetadata.ts

```diff
@@ -12,12 +12,9 @@ export async function setMetadataHandler(context: Context, collection: CE, nft:
     return
   }
 
-  let nftWithToken, existingToken
+  let nftWithToken
```

Comment:
use `const nftWithToken = await getWith(context.store, NE, nft.id, { token: true })`?

Reply:
i'm using it again outside the `try/catch` scope
and the try catch is there because fetching it with token could potentially fail

Reply:
I would move it inside try catch, maybe move `const tokenAPI = new TokenAPI(context.store)` up, and change https://github.com/kodadot/stick/blob/4bc35302b9d7b45b405a40e85674ed8e82ee25d5/src/mappings/shared/token/setMetadata.ts#L25 to check `nftWithToken`, wdyt?

Reply:
done